atlasdb-config/src/main/java/com/palantir/atlasdb/http/FeignOkHttpClients.java

```diff
@@ -105,14 +105,15 @@ private static Client newOkHttpClient(
             Optional<SSLSocketFactory> sslSocketFactory,
             String userAgent,
             boolean retryOnConnectionFailure) {
-        com.squareup.okhttp.OkHttpClient client = new com.squareup.okhttp.OkHttpClient();
-
-        client.setConnectionSpecs(CONNECTION_SPEC_WITH_CYPHER_SUITES);
-        client.setConnectionPool(new ConnectionPool(CONNECTION_POOL_SIZE, KEEP_ALIVE_TIME_MILLIS));
-        client.setSslSocketFactory(sslSocketFactory.orNull());
-        client.setRetryOnConnectionFailure(retryOnConnectionFailure);
-        client.interceptors().add(new UserAgentAddingInterceptor(userAgent));
-        return new OkHttpClient(client);
+        okhttp3.OkHttpClient.Builder builder = new okhttp3.OkHttpClient.Builder()
+                .connectionSpecs(CONNECTION_SPEC_WITH_CYPHER_SUITES)
+                .connectionPool(new ConnectionPool(CONNECTION_POOL_SIZE, KEEP_ALIVE_TIME_MILLIS, TimeUnit.MILLISECONDS))
+                .retryOnConnectionFailure(retryOnConnectionFailure);
+        if (sslSocketFactory.isPresent()) {
+            builder.sslSocketFactory(sslSocketFactory.get());
```

Comment:
Are we able to use http-remoting's `ssl-config` module for the SSL config?
https://github.com/palantir/http-remoting/blob/develop/ssl-config/src/main/java/com/palantir/remoting2/config/ssl/TrustContext.java is basically what you describe for `SSLSocketFactory` & `X509TrustManager`, and see also https://github.com/palantir/http-remoting/blob/develop/ssl-config/src/main/java/com/palantir/remoting2/config/ssl/SslSocketFactories.java